extensions/eclipse-che-theia-plugin-remote/src/node/plugin-remote-init.ts

```diff
@@ -104,9 +118,22 @@ export class PluginRemoteInit {
 
         // bind local stuff
         inversifyContainer.load(pluginRemoteBackendModule);
-
+        inversifyContainer.load(new ContainerModule(bind => {
+            bind(CliManager).toSelf().inSingletonScope();
+            bindContributionProvider(bind, CliContribution);
+            this.bindLogger(bind);
+        }));
         inversifyContainer.bind<number>('plugin.port').toConstantValue(this.pluginPort);
 
+        if (argv === undefined) {
+            argv = process.argv;
+        }
+
+        const cliManager = inversifyContainer.get(CliManager);
+        await cliManager.initializeCli(argv!);
```

Comment:
looks like a hack ?

Reply:
Why?

Reply:
well, first when we try to use container.get() there is something wrong with dependency injection logic

Then we don't need yargs or cli dependency there.

logic is brought forward by the main theia instance, remote is a delegator.

so if log configuration is defined, it's not through cli.

Reply:
No, container.get is necessary because plugin-remote-init.ts itself is not created by the inversion container.